src/Microsoft.CodeQuality.Analyzers/Core/Maintainability/UseNameofInPlaceOfString.cs

```diff
@@ -0,0 +1,152 @@
+        private static readonly LocalizableString s_localizableMessage = new LocalizableResourceString(nameof(MicrosoftMaintainabilityAnalyzersResources.UseNameOfInPlaceOfStringMessage), MicrosoftMaintainabilityAnalyzersResources.ResourceManager, typeof(MicrosoftMaintainabilityAnalyzersResources));
+        private static readonly LocalizableString s_localizableDescription = new LocalizableResourceString(nameof(MicrosoftMaintainabilityAnalyzersResources.UseNameOfInPlaceOfStringDescription), MicrosoftMaintainabilityAnalyzersResources.ResourceManager, typeof(MicrosoftMaintainabilityAnalyzersResources));
+
+        internal static DiagnosticDescriptor RuleWithSuggestion = new DiagnosticDescriptor(RuleId,
+                                                                         s_localizableTitle,
+                                                                         s_localizableMessage,
+                                                                         DiagnosticCategory.Maintainability,
+                                                                         DiagnosticHelpers.DefaultDiagnosticSeverity,
+                                                                         isEnabledByDefault: true,
+                                                                         description: s_localizableDescription,
+                                                                         helpLinkUri: "https://github.com/dotnet/roslyn-analyzers/blob/master/src/Microsoft.CodeQuality.Analyzers/Microsoft.CodeQuality.Analyzers.md#maintainability",
+                                                                         customTags: WellKnownDiagnosticTags.Telemetry);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(RuleWithSuggestion);
+
+        public override void Initialize(AnalysisContext analysisContext)
+        {
+            analysisContext.EnableConcurrentExecution();
+            analysisContext.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            analysisContext.RegisterOperationAction(AnalyzeArgument, OperationKind.Argument);
+        }
+
+        private void AnalyzeArgument(OperationAnalysisContext context)
+        {
+            var argument = (IArgumentOperation)context.Operation;
+            if ((argument.Value.Kind != OperationKind.Literal 
+                || argument.Value.Type.SpecialType != SpecialType.System_String))
+            {
+                return;
+            }
+
+            if (argument.Parameter == null)
+            {
+                return;
+            }
+
+            var stringText = (string)argument.Value.ConstantValue.Value;
+            var properties = ImmutableDictionary<string, string>.Empty.Add(StringText, stringText);
+
+            var matchingParameter = argument.Parameter;
+
+            switch (matchingParameter.Name)
+            {
+                case ParamName:
```

Comment:
sorry I dont get this one. so it only work if parameter name is named paramName or propertyName?

Reply:
so, this only works for method that has parameter that is named paramName and propertyName. that is a bit to strict feels like... but anyway, I see where this is intended to be used. so. ignore my comment.

Reply:
That is correct, we're targeting scenarios where we know nameof can and should be used.  Issue #1421 has been opened to consider other parameter names.